src/cc_bstring.c

```diff
@@ -142,3 +142,41 @@ bstring_atou64(uint64_t *u64, struct bstring *str)
 
     return CC_OK;
 }
+
+struct bstring *
+bstring_alloc(uint32_t size)
+{
+    struct bstring bs;
+    bstring_init(&bs);
+
+    bs.len = size;
+    bs.data = cc_alloc(size);
+    if (bs.data == NULL) {
+        return NULL;
+    }
+
+    struct bstring *rv = cc_alloc(sizeof(struct bstring));
+    if (rv == NULL) {
+        free(bs.data);
+        return NULL;
+    }
+
+    memcpy(rv, &bs, sizeof(struct bstring));
```

Comment:
why not just start by allocating struct bstring to the pointer and initializing/allocating the underlying buffer there, instead of copying from this bstring on the stack?

Reply:
So my thought was that if the `cc_alloc` fails, I don't have to clean anything up, I can just return NULL and don't have to worry about freeing the pointer. I can change it if this isn't idiomatic.

Reply:
I think allocating the pointer and initializing that directly results in the same number of failure clean ups. Like this:
```
struct bstring *bs = cc_alloc(sizeof(*bs));
if (bs == NULL) {
    return NULL;
}

bs.len = size;
bs.data = cc_alloc(size);
if (bs.data == NULL) {
    cc_free(bs);
    return NULL;
}
```